### ISSUE TITLE
VC++ release builds don't undefine NDEBUG; asserts are ignored.

### ISSUE THREAD

**Issue by lukester1975** [WebAssembly/binaryen#874]
Hi

VC++ and non-VC++ release builds differ in their assert behavior.

asm2wasm.cpp VC++ command line (Release build):

<code>/GS /TP /analyze- /W3 /wd"4146" /Zc:wchar_t /I"C:\Users\luke\.emscripten_ports\binaryen\binaryen-version_23\src" /I"C:\WpdPack\Include" /I"C:\OpenSSL-Win32\include" /Gm- /O2 /Ob2 /Fd"asm2wasm.dir\Release\vc140.pdb" /Zc:inline /fp:precise /D "WIN32" /D "_WINDOWS" /D "_CRT_SECURE_NO_WARNINGS" /D "_SCL_SECURE_NO_WARNINGS" /D "NDEBUG" /D "CMAKE_INTDIR=\"Release\"" /D "_MBCS" /errorReport:prompt /WX- /Zc:forScope /GR /Gd /Oy- /MD /Fa"Release/" /EHsc /nologo /Fo"asm2wasm.dir\Release\" /Fp"asm2wasm.dir\Release\asm2wasm.pch"
</code>

vs CMakeFilerds/asm2wasm.dir/flags.make:

<code>CXX_FLAGS =  -std=c++11 -msse2 -mfpmath=sse -Wall -Werror -Wextra -Wno-unused-parameter -fno-omit-frame-pointer -fPIC -O2 -DNDEBUG -O2 -UNDEBUG   -std=gnu++11
</code>

I guess the MSVC case in CMakeLists.txt needs:

<code>ADD_NONDEBUG_COMPILE_FLAG("/UNDEBUG")</code>

(assuming cmake is smart enough to put it after the /DNDEBUG...)

I found this due to tripping over an assert with GCC built asm2wasm but not with a VC++ built asm2wasm - I'll create a separate ticket for the actual assert.

Best regards

Luke.

**Comment by lukester1975** [comment]
Hi

Looks like this can be closed after #882.

Thanks!

Luke.
